Marlin/src/lcd/dwin/e3v2/dwin.cpp

```diff
@@ -1800,38 +1835,50 @@ void HMI_SDCardUpdate() {
-  #endif
+  DWIN_ICON_Show(ICON, ICON_BedTemp, 10, 416);
+  DWIN_Draw_IntValue(true, true, 0, DWIN_FONT_STAT, Color_White, Color_Bg_Black, 3, 28, 417, thermalManager.temp_bed.celsius);
+  DWIN_Draw_String(false, false, DWIN_FONT_STAT, Color_White, Color_Bg_Black, 25 + 3 * STAT_CHR_W + 5, 417, (char*)"/");
+  DWIN_Draw_IntValue(true, true, 0, DWIN_FONT_STAT, Color_White, Color_Bg_Black, 3, 25 + 4 * STAT_CHR_W + 6, 417, thermalManager.temp_bed.target);
 
-  DWIN_ICON_Show(ICON, ICON_Speed, 13, 429);
-  DWIN_Draw_IntValue(true, true, 0, DWIN_FONT_STAT, Color_White, Color_Bg_Black, 3, 33 + 2 * STAT_CHR_W, 429, feedrate_percentage);
-  DWIN_Draw_String(false, false, DWIN_FONT_STAT, Color_White, Color_Bg_Black, 33 + 5 * STAT_CHR_W + 2, 429, F("%"));
+  DWIN_ICON_Show(ICON, ICON_Speed, 113, 383);
+  DWIN_Draw_IntValue(true, true, 0, DWIN_FONT_STAT, Color_White, Color_Bg_Black, 3, 116 + 2 * STAT_CHR_W, 384, feedrate_percentage);
+  DWIN_Draw_String(false, false, DWIN_FONT_STAT, Color_White, Color_Bg_Black, 116 + 5 * STAT_CHR_W + 2, 384, (char*)"%");
 
-  #if HAS_ZOFFSET_ITEM
-    DWIN_ICON_Show(ICON, ICON_Zoffset, 158, 428);
-    dwin_zoffset = BABY_Z_VAR;
-    DWIN_Draw_Signed_Float(DWIN_FONT_STAT, Color_Bg_Black, 2, 2, 178, 429, dwin_zoffset * 100);
-  #endif
+  DWIN_ICON_Show(ICON, ICON_StepE, 112, 417);
+  DWIN_Draw_IntValue(true, true, 0, DWIN_FONT_STAT, Color_White, Color_Bg_Black, 3, 116 + 2 * STAT_CHR_W, 417, flowrate);
+  DWIN_Draw_String(false, false, DWIN_FONT_STAT, Color_White, Color_Bg_Black, 116 + 5 * STAT_CHR_W + 2, 417, (char*)"%");
+
+  DWIN_ICON_Show(ICON, ICON_FanSpeed, 187, 383);
+  DWIN_Draw_IntValue(true, true, 0, DWIN_FONT_STAT, Color_White, Color_Bg_Black, 3, 195 + 2 * STAT_CHR_W, 384, thermalManager.fan_speed[0]);
+
+  DWIN_ICON_Show(ICON, ICON_Zoffset, 187, 416);
+
+  zoffsetvalue = BABY_Z_VAR; // Temporary Statement for Compatability
+
+  if (zoffsetvalue < 0) {
+    DWIN_Draw_FloatValue(true, true, 0, DWIN_FONT_STAT, Color_White, Color_Bg_Black, 2, 2, 207, 417, -zoffsetvalue * 100);
+    DWIN_Draw_String(false, true, font8x16, Color_White, Color_Bg_Black, 205, 419, (char*)"-");
+  }
+  else {
+    DWIN_Draw_FloatValue(true, true, 0, DWIN_FONT_STAT, Color_White, Color_Bg_Black, 2, 2, 207, 417, zoffsetvalue * 100);
+    DWIN_Draw_String(false, true, font8x16, Color_White, Color_Bg_Black, 205, 419, (char*)" ");
+  }
+
+  DWIN_Draw_Rectangle(1, Line_Color, 0, 449, DWIN_WIDTH, 451);
+
+  DWIN_ICON_Show(ICON, ICON_MaxSpeedX,   10, 456);
+  DWIN_Draw_FloatValue(true, true, 0, font8x16, Color_White, Color_Bg_Black, 3, 1, 35, 459, current_position.x * 10);
+  DWIN_ICON_Show(ICON, ICON_MaxSpeedY,   95, 456);
+  DWIN_Draw_FloatValue(true, true, 0, font8x16, Color_White, Color_Bg_Black, 3, 1, 120, 459, current_position.y * 10);
+  DWIN_ICON_Show(ICON, ICON_MaxSpeedZ,   180, 456);
+  DWIN_Draw_FloatValue(true, true, 0, font8x16, Color_White, Color_Bg_Black, 3, 1, 205, 459, current_position.z * 10);
```

Comment:
Actual numbers should only be shown when the printer is homed - ie it is accurate. Currently, these will show as something like 0 or something different for Z position.

I believe its standard to use a '?' as a value to display if the current status of the printer is not homed.

As an example: https://github.com/MarlinFirmware/Marlin/blob/8e1ea6a2fa1b90a58b4257eec9fbc2923adda680/Marlin/src/lcd/tft/ui_480x320.cpp#L272

Reply:
The only issue I find with such a convention is that most of the time the printer does not actually know if it is currently homed or not, ie if the steppers are disabled and moved by hand. Given these cases I feel as though such a convention might give the false illusion that the printer is homed anytime the numbers are displayed. This is just my opinion though, perhaps it is still best to have it show a '?' even if its just before the first home.

Reply:
Correct. Unless the steppers are always engaged after a home, the printer switches to using `is_homed` = false.

As such, you can display a static ? unless `is_homed` is true.

Reply:
I added the requested fix, but as far as I can tell, disabling the steppers does not designate that the axis are not homed. As I said before, this means once the printer has been homed the printer will have no idea from that point on whether or not it is still homed. Maybe I'm missing something, but that is what I could deduce from the code and from my testing.

Reply:
I may have found the issue. I could use the "trusted" parameter of the axis rather than the "homed" parameter. As it does get reset on M84 commands, but I am unsure if this would lead to any other issues...

Reply:
I'm happy for this to be reviewed by @thinkyhead and the higher ups in the process!

Nice work.